src/programs/federal/snap.js

```diff
@@ -12,30 +12,32 @@ import { getLimitBySize } from '../../utils/getGovData';
 import { toMonthlyFrom } from '../../utils/convert-by-timescale';
 import {
   getEveryMemberOfHousehold,
-  getEveryMember,
-  getDependentsOfHousehold,
+  getDependentMembers,
   isDisabled,
-  isUnder13,
-  getUnder13OfHousehold,
+  getYoungerThan,
+  getOlderThan,
 } from '../../utils/getMembers';
 
 
-/** Based on https://www.masslegalservices.org/SNAPCalculator */
-const getSNAPBenefits = function (client, timeframe) {
+/** Based on https://www.masslegalservices.org/SNAPCalculator
+ * @note Bay State CAP not included as this prototype only deals with
+ *     changes in earned income */
+const getSNAPBenefits = function (fullClient, timeframe) {
 
-  client = client[ timeframe ];
+  var client = fullClient[ timeframe ];
 
   var finalResult = 0,
-      grossIncomeTestResult   = hlp.getGrossIncomeTestResult(client),
-      netIncomeTestResult     = hlp.getNetIncomeTestResult(client),
-      maxSnapAllotment        = getLimitBySize(SNAPData.SNAP_LIMITS, hlp.householdSize(client)),
-      percentageOfNetIncome   = hlp.monthlyNetIncome(client) * SNAPData.PERCENT_OF_NET,
+      householdSize           = hlp.getHouseholdSize(client),
+      grossIncomeTestResult   = hlp.passesGrossIncomeTest(client),
+      netIncomeTestResult     = hlp.passesNetIncomeTest(client),
+      maxSnapAllotment        = getLimitBySize(SNAPData.SNAP_LIMITS, householdSize),
+      percentageOfNetIncome   = hlp.getNetIncome(client) * SNAPData.PERCENT_OF_NET,
       maxClientAllotment      = Math.max(0, maxSnapAllotment - percentageOfNetIncome);
 
   if (grossIncomeTestResult === true && netIncomeTestResult === true) {
 
     if (maxClientAllotment <= SNAPData.SMALL_HOUSEHOLD_MIN_GRANT) {
-      if (hlp.householdSize(client) <= SNAPData.SMALL_HOUSEHOLD_SIZE) {
+      if (householdSize <= SNAPData.SMALL_HOUSEHOLD_SIZE) {
```

Comment:
Currently, with this nested if statement, a household larger than `SMALL_HOUSEHOLD_SIZE` but with a max client allotment of less than `SMALL_HOUSEHOLD_MIN_GRANT` will get a `finalAmount` of 0. If that's a mistake, we should change this to a single if statement with an &&.

Reply:
Yeah true, I've just never read through this file before and thought it seemed important.

Reply:
I don't remember any particular one of these, but I do remember finding several places where the logic didn't make sense to me and ended up matching the mass legal logic. We can take another look at that later for sure. @todo.

Other than that, are more changes needed?